### ISSUE TITLE
Shipping tax fails to calculate when using the backend

### ISSUE THREAD

**Issue by adaddeo** [solidusio/solidus#1288]
Calculating, and re-calculating shipping tax when building an order in the backend is a bit funky. I've listed a flow below that can be performed on a 1.3.0 sandbox to replicate.
1. In the admin, create a new order.
2. Add a Ruby on Rails Bag.
3. Under customer details, fill out any U.S. address and submit.
4. Under shipments, select UPS One Day.
5. Go to payments.

At this point we have an order total of $39.14, which is 75 cents short of what it should be due to the missing shipping tax. This is because the order is still in the delivery state, even though we are trying to create a payment. There is no way intuitive way to have the order transition to payment.
1. Submit a payment with the default amount ($39.14).

A payment is created for a (different) amount that includes the newly created shipping tax ($39.89). This might be a bug: the payment form is submitted with $39.14 in the amount field, but after the order transitions and calculates the shipping tax, a payment is created for $39.89 instead.
1. Return to the shipments page, change the shipping method to UPS Ground.

The shipping tax adjustment changes from 75 cents to 25 cents, but the tax total on the order doesn't get updated. Again, no intuitive way to have the order re-calculate/transition.

**Comment by dangerdogz** [comment]
I ran into something similar yesterday, I think the issue at hand is that taxes are updated by the OrderUpdater and that changing the selected_shipping_method from the backend does not trigger anything that calls the OrderUpdater because it goes through the API and simply update the attributes, while adding a payment trigger @order.next which calls OrderUpdater through a transition callback

from [shipment.rb] (https://github.com/solidusio/solidus/blob/master/core/app/models/spree/shipment.rb)

```ruby
def selected_shipping_rate_id=(id)
   shipping_rates.update_all(selected: false)
   shipping_rates.update(id, selected: true)
   save!
end
```

**Comment by jordan-brough** [comment]
Note to self: We should try to tackle this while working on https://github.com/solidusio/solidus/issues/1252

**Comment by jordan-brough** [comment]
@adaddeo & @dangerdogz thanks for the detailed info.

**Comment by jhawthorn** [comment]
This should be fixed by #1514, which has been merged into master

**Comment by adaddeo** [comment]
Great stuff!
